WcaOnRails/app/javascript/edit-events/modals/index.jsx

```diff
@@ -0,0 +1,151 @@
+import cn from 'classnames'
+import Modal from 'react-bootstrap/lib/Modal'
+import Button from 'react-bootstrap/lib/Button'
+import Checkbox from 'react-bootstrap/lib/Checkbox'
+
+import events from 'wca/events.js.erb'
+import formats from 'wca/formats.js.erb'
+import { rootRender } from 'edit-events'
+
+import CutoffComponents from './Cutoff'
+import TimeLimitComponents from './TimeLimit'
+import AdvancementConditionComponents from './AdvancementCondition'
+import ButtonActivatedModal from 'edit-events/ButtonActivatedModal'
+
+let RoundAttributeComponents = {
+  timeLimit: TimeLimitComponents,
+  cutoff: CutoffComponents,
+  advancementCondition: AdvancementConditionComponents,
+};
+
+function findRoundsSharingTimeLimitWithRound(wcifEvents, wcifRound) {
+  let roundsSharingTimeLimit = [];
+  wcifEvents.forEach(otherWcifEvent => {
+    otherWcifEvent.rounds.forEach(otherWcifRound => {
+      if(otherWcifRound == wcifRound || !otherWcifRound.timeLimit) {
+        return;
+      }
+
+      if(otherWcifRound.timeLimit.cumulativeRoundIds.indexOf(wcifRound.id) >= 0) {
+        roundsSharingTimeLimit.push(otherWcifRound);
+      }
+    });
+  });
+  return roundsSharingTimeLimit;
+}
+
+function findRounds(wcifEvents, roundIds) {
+  let wcifRounds = [];
+  wcifEvents.forEach(wcifEvent => {
+    wcifEvent.rounds.forEach(wcifRound => {
+      if(roundIds.indexOf(wcifRound.id) >= 0) {
+        wcifRounds.push(wcifRound);
+      }
+    });
+  });
+  return wcifRounds;
+}
+
+class EditRoundAttribute extends React.Component {
+  componentWillMount() {
+    this.reset();
+  }
+
+  componentWillReceiveProps() {
+    this.reset();
+  }
+
+  getWcifRound() {
+    let { wcifEvent, roundNumber } = this.props;
+    return wcifEvent.rounds[roundNumber - 1];
+  }
+
+  getSavedValue() {
+    return this.getWcifRound()[this.props.attribute];
+  }
+
+  hasUnsavedChanges = () => {
+    return JSON.stringify(this.getSavedValue()) != JSON.stringify(this.state.value);
+  }
+
+  onChange = (value) => {
+    this.setState({ value: value });
+  }
+
+  onOk = () => {
+    let wcifRound = this.getWcifRound();
+    wcifRound[this.props.attribute] = this.state.value;
+
+    // This is gross. timeLimit is special because of cross round cumulative time limits.
+    // If you set a time limit for 3x3x3 round 1 shared with 2x2x2 round 1, then we need
+    // to make sure the same timeLimit gets set for both of the rounds.
+    if(this.props.attribute == "timeLimit") {
+      let timeLimit = this.state.value;
+
+      // First, remove this round from all other rounds that previously shared
+      // a time limit with this round.
+      findRoundsSharingTimeLimitWithRound(this.props.wcifEvents, wcifRound).forEach(otherWcifRound => {
+        let index = otherWcifRound.timeLimit.cumulativeRoundIds.indexOf(wcifRound.id);
+        if(index < 0) {
+          throw new Error();
+        }
+        otherWcifRound.timeLimit.cumulativeRoundIds.splice(index, 1);
+      });
```

Comment:
I think this could simply be:
```js
findRoundsSharingTimeLimitWithRound(this.props.wcifEvents, wcifRound).forEach(otherWcifRound => {
  _.pull(otherWcifRound.timeLimit.cumulativeRoundIds, wcifRound.id);
});
```